### PR TITLE
Clean up unused code

### PR DESCRIPTION
**Pull request by KelvinLi**
Simply clean up some unused variables and import statements.

Thanks to Thomas Caswell for pointing out potential conflicts with PR #4143 and PR #3421 (see http://sourceforge.net/p/matplotlib/mailman/message/33920364/). I've backed out the WX backend changes, and Git seems to happily auto merge with them now.

### CODE REVIEW

File: lib/matplotlib/backends/backend_ps.py [hunk 1]
```diff
@@ -27,36 +21,24 @@ def _fn_name(): return sys._getframe(1).f_code.co_name
...
 from matplotlib import _path
 from matplotlib.transforms import Affine2D
-
 from matplotlib.backends.backend_mixed import MixedModeRenderer
 
-
-import numpy as np
-import binascii
-import re
 try:
```
Comment (jenshnielsen) [inline lib/matplotlib/backends/backend_ps.py]: This try except can go too. All versions of python that we support have set in the global namespace

File: lib/matplotlib/backends/backend_agg.py [hunk 2]
```diff
@@ -22,8 +22,6 @@
 from __future__ import (absolute_import, division, print_function,
                         unicode_literals)
 
-import six
```
Comment (tacaswell) [inline lib/matplotlib/backends/backend_agg.py]: Please leave six everywhere.  Current policy is every file gets the future import + six.

File: lib/matplotlib/backends/backend_qt4.py [hunk 3]
```diff
@@ -1,48 +1,16 @@
...
-from six import unichr
-import os
-import re
-import signal
-import sys
-
-import matplotlib
-
-from matplotlib.cbook import is_string_like
-from matplotlib.backend_bases import FigureManagerBase
```
Comment (tacaswell) [inline lib/matplotlib/backends/backend_qt4.py]: These all probobly need to stay for back compatibility reasons.

File: lib/matplotlib/backends/backend_qt4.py [hunk 4]
```diff
@@ -1,48 +1,16 @@
...
-from matplotlib.widgets import SubplotTool
-try:
-    import matplotlib.backends.qt_editor.figureoptions as figureoptions
-except ImportError:
-    figureoptions = None
-
-from .qt_compat import QtCore, QtWidgets, _getSaveFileName, __version__
-from matplotlib.backends.qt_editor.formsubplottool import UiSubplotTool
-
-from .backend_qt5 import (backend_version, SPECIAL_KEYS, SUPER, ALT, CTRL,
```
Comment (tacaswell) [inline lib/matplotlib/backends/backend_qt4.py]: Same with these, most of these import are there to preserve the API even though we re-organized things.

Comment (KelvinLi) [inline lib/matplotlib/backends/backend_qt4.py]: Thanks for the tips. Just for my own understanding, does this mean we cannot break users who do the following?

`from matplotlib.backends.backend_qt4 import SubplotTool`

I'm surprised that this is considered part of the external API, but I'll take your word for it.

Assuming we are on the same page, then moving forward, I'm not even sure what is considered safe to prune... any advice?

Comment (tacaswell) [inline lib/matplotlib/backends/backend_qt4.py]: @efiring @mdboom thoughts?  This issue is why I have thought about but
never done this pruning.

At a minimum all the removals need to be documented as api changes with
where to import them from instead.

The gui backends are heavily used by people embedding in larger
applications so many strange things may be going on.

The problem is that I don't think exactly what is part of the public api
has been well documented, anything with out a leading _ is considered
public in python and we err on the side of not being user code unless there
is a very compelling reason.

Comment (WeatherGod) [inline lib/matplotlib/backends/backend_qt4.py]: If there is an __all__ in the module, then I would say that it is safe to
throw out anything not listed in the __all__ that isn't needed anymore.
But, yeah, this is what happens when you become a major package in a
language and you aren't careful when you create modules. You learn the
lesson and try and do better moving forward.

Comment (efiring) [inline lib/matplotlib/backends/backend_qt4.py]: Maybe that's the way to go--use `__all__` to declare the API.  There's a lot of stuff that we don't want to carry around forever.

Comment (tacaswell) [inline lib/matplotlib/backends/backend_qt4.py]: Even adding `__all__` can break code for people if it does not include
everything they are relying on.

That said, if you are blindly using import *, I don't feel _too_ bad for
you when it gets broken.

Comment (KelvinLi) [inline lib/matplotlib/backends/backend_qt4.py]: There are only a few modules that are declared safe for `import *`, correct? I believe `pylab` and `pyplot` are such examples (?)

So I'll try a new approach and annotate everything with `__all__`.

Thanks for `__all__` the feedback!

**Comment by OceanWolf** [comment]
I will feel a lot happier once MEP27 lands, so that we then have a solid base to work from.

**Comment by KelvinLi** [comment]
I've opened a new request, #4486, to replace this one

**Pull request closed without merging**